src/services/repo/useRepoCoverage.spec.tsx

```diff
@@ -0,0 +1,243 @@
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
+import { renderHook, waitFor } from '@testing-library/react'
+import { graphql } from 'msw'
+import { setupServer } from 'msw/node'
+
+import { useRepoCoverage } from './useRepoCoverage'
+
+const mockRepoCoverage = {
+  owner: {
+    repository: {
+      __typename: 'Repository',
+      branch: {
+        name: 'main',
+        head: {
+          yamlState: 'DEFAULT',
+          totals: {
+            percentCovered: 70.44,
+            lineCount: 90,
+            hitsCount: 80,
+          },
+        },
+      },
+    },
+  },
+}
+
+const mockNullBranch = {
+  owner: {
+    repository: {
+      __typename: 'Repository',
+      branch: null,
+    },
+  },
+}
+
+const mockRepoNotFound = {
+  owner: {
+    repository: {
+      __typename: 'NotFoundError',
+      message: 'repo not found',
+    },
+  },
+}
+
+const mockOwnerNotActivated = {
+  owner: {
+    repository: {
+      __typename: 'OwnerNotActivatedError',
+      message: 'owner not activated',
+    },
+  },
+}
+
+const queryClient = new QueryClient({
+  defaultOptions: { queries: { retry: false } },
+})
+
+const wrapper: React.FC<React.PropsWithChildren> = ({ children }) => (
+  <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
+)
+
+const server = setupServer()
+
+console.error = () => {}
```

Comment:
Can we mock this out in the tests where it's needed, and reset it afterwards, there can be some ... interesting behaviour by mocking it out at the top level, also can be useful if for whatever reason the tests that are following the happy path fail, this can help narrow down why they're failing